src/ray/protobuf/gcs.proto

```diff
@@ -103,15 +103,17 @@ message TaskTableData {
 message ActorTableData {
   // State of an actor.
   enum ActorState {
-    // Actor is pending.
-    PENDING = 0;
+    // Actor dependencies are resolving.
+    PENDING_DEPENDENCY_RESOLUTION = 0;
```

Comment:
How about `DEPENDENCY_UNREADY`? This sounds clearer to me.
@wumuzi520 @rkooo567 @stephanie-wang 

If you agree with this, we should also rename `ReportActorDependenciesResoloved` to `ReportActorDependenciesReady`.

Reply:
Both sounds fine to me. But, I do think `ReportActorDependenciesReady` is the misleading name. This is confusing for a caller that this RPC callback is called after dependency is reported (which doesn't seem to be true)

Reply:
@rkooo567 Actually, I think it's better to make the RPC reply right after the dependencies are reported. So GCS server don't need to hold the pending requests (and can save memory and number of connections). 
We used to do that in the beginning. However, that's is not compatible with ref counting. So we ended up changing it this way. 
We can look into the ref counting issue after this PR.

Reply:
I will put the response to this here https://github.com/ray-project/ray/pull/8045#discussion_r459211368. 

### Response to the message:
Firstly, I don't think the optimization is necessary. It seems to give really tiny benefit compared to the complexity. So, I will not think about optimization stuff here.

Imo, from core worker's perspective, it doesn't need to be aware of any step of actor creation. I think it is pretty clear to understand what's going on if we 1 Register actor before dependencies are resolved. 2. send a creation request after dependencies are resolved. 1~2 lines comments about why we need 2 steps looks sufficient to understand what it is doing. Calling it `ReportActorDependenciesReady ` just adds unnecessary mental overhead (because caller doesn't know when actor will be actually created without knowing the implementation details about GCS actor management), especially when the callback is called after actors are actually created. 

>I think it's better to make the RPC reply right after the dependencies are reported.
In this case, I have no objection to the name. But, if it causes any ref counting issues, it should be handled inside this PR, especially given we will have a branch cut really soon. We should not merge the PR that will bring stability regression.

Reply:
Btw, I don't think we should spend so much time deciding this name. But, as a reader perspective, `ReportActorDependenciesReady` doesn't seem to be the right name in this context. (unless you make it reply after dependencies reports are done in GCS)

Reply:
I would like to fix the reply timing issue ASAP. However, that would add extra complexity to this PR and would delay it. So I prefer to fix that in the next PR, as that issue isn't as urgent as the actor task hanging issue.

I really don't think the timing of reply should impact the naming of a request. The most important thing about the name is to reflect the purpose of the method. If you do care about the timing of the reply, we can add a comment there to avoid confusion.

Reply:
Hmm.. I don't know how the timing of reply doesn't matter. If the reply comes after actor is created, this RPC is "creating an actor" not "reporting dependencies". It is just the wrong name that exposes implementation details. It seems to me that you define your function name based on the first line of the function body.

Reply:
Also, Idk how it exactly breaks the ref counting protocol, but if so, why don't you just add all of them in the next PR? You can just make another PR that 

1. Makes this RPC reply after dependencies are reported.
2. And fixes the ref counting protocol problems caused by that. 

I don't see any compelling reasons to add 1 into this PR, break ref counting protocol, and make another PR to fix it.

Reply:
@rkooo567 I think it makes sense to rename the `ReportActorDependenciesReady`  to `CreateActor`, I will do it now.

Reply:
BTW, does the state `PENDING_DEPENDENCY_RESOLUTION` need to be renamed too? @rkooo567 @raulchen

Reply:
okay, we can rename it later if you insist.

Reply:
I prefer `DEPENDENCY_UNREADY`.

Reply:
I don't have a strong preference. `DEPENDENCY_UNREADY` seems fine (but I would probably call it `DEPENDENCIES_UNREADY` to be more accurate).

Reply:
Same. I don't have any preference. `DEPENDECIES_UNREADY` or `DEPENDENCY_READY` seems to be fine!